File: 2026-2/sesion-de-clase-3/lab01-2026-1/QuestionManager.java

import java.util.ArrayList;
import java.util.Collections;
import java.util.List;

public class QuestionManager {

    private List<Question> preguntas;

    public QuestionManager() {
        this.preguntas = new ArrayList<>();
    }

    public List<Question> getPreguntas() {
        return Collections.unmodifiableList(preguntas);
    }

    public void add(Question question) {
        preguntas.add(question);
    }

    public List<Question> selectRandomN(int cantidad) {
        List<Question> copia = new ArrayList<>(this.preguntas);
        Collections.shuffle(copia);
        return copia.subList(0, Math.min(cantidad, copia.size()));
    }
}

File: 2026-2/sesion-de-clase-05/dotnet/Lab01Lib/QuestionManager.cs
namespace PUCP.Labs.Lab01.Admin;

using PUCP.Labs.Lab01.Preguntas;

public class QuestionManager
{
    private readonly List<Question> preguntas = new List<Question>();

    public IList<Question> Preguntas => preguntas.AsReadOnly();

    public void Add(Question question)
    {
        preguntas.Add(question);
    }

    public List<Question> SelectRandomN(int cantidad)
    {
        List<Question> copia = [.. preguntas];
        Random random = new();
        int n = copia.Count;
        while (n > 1)
        {
            n--;
            int k = random.Next(n + 1);
            (copia[k], copia[n]) = (copia[n], copia[k]);
        }
        return copia.GetRange(0, Math.Min(cantidad, copia.Count));
    }
}
